src/django/api/views.py

```diff
@@ -3263,3 +3309,110 @@ def retrieve(self, request, pk=None):
     def update(self, request, pk=None):
         self.validate_request(request)
         return super(ApiBlockViewSet, self).update(request, pk=pk)
+
+
+class FacilityActivityReportAutoSchema(AutoSchema):
+    def get_serializer_fields(self, path, method):
+        return [
+            coreapi.Field(
+                name='data',
+                location='body',
+                description=(
+                    'The country, name, and address of the facility. See '
+                    'the sample request body above.'),
+                required=True,
+            )
+        ]
+
+
+def update_facility_activity_report_status(facility_activity_report,
+                                           request, status):
+    status_change_reason = request.data.get('status_change_reason')
+    now = str(datetime.utcnow())
+
+    facility_activity_report.status_change_reason = status_change_reason
+    facility_activity_report.status_change_by = request.user
+    facility_activity_report.status_change_date = now
+    facility_activity_report.status = status
+    if status == 'CONFIRMED':
+        facility_activity_report.approved_at = now
+    facility_activity_report.save()
+
+    return facility_activity_report
+
+
+@schema(FacilityActivityReportAutoSchema())
+class FacilityActivityReportViewSet(viewsets.GenericViewSet):
+    """
+    Approve or reject FacilityActivityReports.
+
+    ## Sample Request Body
+
+        {
+            "status_change_reason": "CLOSED"
+        }
+    """
+    queryset = FacilityActivityReport.objects.all()
+    serializer_class = FacilityActivityReportSerializer
+    permission_classes = (IsRegisteredAndConfirmed,)
+
+    @action(detail=True, methods=['PUT'],
+            permission_classes=(IsRegisteredAndConfirmed,),
```

Comment:
We should change this permission class to `IsAdminUser` since we don't want to allow people to approve their own reports.